src/nvim/msgpack_rpc/channel.c

```diff
@@ -306,8 +306,11 @@ static void receive_msgpack(Stream *stream, RBuffer *rbuf, size_t c, void *data,
   p->read_ptr = rbuffer_read_ptr(rbuf, &size);
   p->read_size = size;
   parse_msgpack(channel);
-  size_t consumed = size - p->read_size;
-  rbuffer_consumed_compact(rbuf, consumed);
+
+  if (!unpacker_closed(p)) {
+    size_t consumed = size - p->read_size;
+    rbuffer_consumed_compact(rbuf, consumed);
```

Comment:
Although @KillTheMule says this doesn't fix https://github.com/neovim/neovim/issues/23781#issuecomment-1902176915 , checking `unpacker_closed` still looks like it's needed because there is nothing else in `channel.c` to stop `receive_msgpack` from being called if the unpacker is closed...

Do we need to call `chan_close_with_error` as we do above?